src/main/kotlin/ecommerce/service/PaymentService.kt

```diff
@@ -0,0 +1,69 @@
+package ecommerce.service
+
+import ecommerce.client.StripeClient
+import ecommerce.dto.OrderRequest
+import ecommerce.dto.PaymentResponse
+import ecommerce.enum.PaymentStatus
+import ecommerce.exception.StripePaymentException
+import ecommerce.model.Payment
+import ecommerce.repository.PaymentJpaRepository
+import ecommerce.repository.getByPaymentIntentIdOrThrow
+import jakarta.transaction.Transactional
+import org.hibernate.service.spi.ServiceException
+import org.springframework.stereotype.Service
+import java.math.BigDecimal
+
+@Transactional
+@Service
+class PaymentService(
+    private val stripeClient: StripeClient,
+    private val paymentJpaRepository: PaymentJpaRepository,
+) {
+    fun createPaymentIntent(
+        request: OrderRequest,
+        amount: BigDecimal,
+    ): PaymentResponse {
+        try {
+            val paymentResponse =
+                stripeClient.createCheckoutSession(request, amount.multiply(BigDecimal.valueOf(100)).toInt())
+                    ?: throw ServiceException("Payment Failed")
+
+            val payment =
+                Payment(
+                    amount = amount,
+                    currency = request.currency,
+                    paymentMethod = request.paymentMethod,
+                    paymentIntentId = paymentResponse.id,
+                )
+            paymentJpaRepository.save(payment)
+            return paymentResponse
+        } catch (e: StripePaymentException) {
+            val paymentResponse =
+                PaymentResponse(
+                    id = "Not Available",
+                    amount = amount.toDouble(),
+                    status = PaymentStatus.FAILED.name,
+                    errorMessage = e.message,
+                )
+            val payment =
+                Payment(
+                    amount = amount,
+                    currency = request.currency,
+                    paymentMethod = request.paymentMethod,
+                    paymentIntentId = "Payment Intent Failed",
+                    status = PaymentStatus.FAILED,
+                )
+            paymentJpaRepository.save(payment)
+            return paymentResponse
+        }
```

Comment:
I see you've `return`ed a response in a catch statement rather than throwing a new exception 😮 
This is to prevent rollback from happening right? So we can persist failed payment?